src/fold/transformations/lags.py

```diff
@@ -54,10 +55,19 @@ def __init__(
 
     def transform(self, X: pd.DataFrame, in_sample: bool) -> pd.DataFrame:
         X = X.copy()
-        for column, lags in self.columns_and_lags:
-            lags = wrap_in_list(lags)
-            for lag in lags:
-                X[f"{column}_lag_{lag}"] = X[column].shift(lag)[-len(X) :]
+
+        if is_columns_all([self.columns_and_lags[0][0]]):
+            lags = wrap_in_list(self.columns_and_lags[0][1])
+            X = pd.concat(
+                [X]
+                + [X.shift(lag)[-len(X) :].add_suffix(f"_lag_{lag}") for lag in lags],
+                axis="columns",
+            )
+        else:
+            for column, lags in self.columns_and_lags:
+                lags = wrap_in_list(lags)
+                for lag in lags:
+                    X[f"{column}_lag_{lag}"] = X[column].shift(lag)[-len(X) :]
```

Comment:
could you add a single test case for both AddLagsX and AddWindowFeatures, to check if everything works as expected?

Reply:
Yes, done!